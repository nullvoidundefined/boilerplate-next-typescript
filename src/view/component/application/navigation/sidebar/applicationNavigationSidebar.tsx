import styled from "styled-components";

import {
  APPLICATION_COLOR,
  APPLICATION_DIMENSION,
} from "../../../../../constant";

interface ApplicationNavigationSidebarProps {
  heightOffset: number;
}

const StyledApplicationNavigationSidebar = styled.div<ApplicationNavigationSidebarProps>`
  background-color: ${APPLICATION_COLOR.SIDEBAR_BACKGROUND};
  max-width: ${APPLICATION_DIMENSION.SIDEBAR_WIDTH_DESKTOP}px;
  min-height: ${({ heightOffset }) => {
    return `calc(100vh - ${heightOffset}px)`;
  }};
  width: ${APPLICATION_DIMENSION.SIDEBAR_WIDTH_DESKTOP}px;
`;

const ApplicationNavigationSidebar = ({ heightOffset }: ApplicationNavigationSidebarProps) => {
  return (
    <StyledApplicationNavigationSidebar heightOffset={heightOffset}>
      Sidebar
    </StyledApplicationNavigationSidebar>
  );
};

export { ApplicationNavigationSidebar };
