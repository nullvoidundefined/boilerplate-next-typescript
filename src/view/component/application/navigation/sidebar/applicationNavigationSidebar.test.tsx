import "jest-styled-components";

import { ApplicationNavigationSidebar } from "./applicationNavigationSidebar";
import renderer from "react-test-renderer";
import { noop } from "lodash";

it("renders correctly", () => {
  const props = {
    heightOffset: 0,
  };
  const component = renderer
    .create(<ApplicationNavigationSidebar {...props} />)
    .toJSON();
  expect(component).toMatchSnapshot();
});
