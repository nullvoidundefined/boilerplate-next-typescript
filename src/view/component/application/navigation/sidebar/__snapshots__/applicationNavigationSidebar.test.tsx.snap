// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c0 {
  background-color: green;
  max-width: 160px;
  min-height: calc(100vh - 0px);
  width: 160px;
}

<div
  className="c0"
>
  Sidebar
</div>
`;
